lib/dialects/mssql/blocks.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+var _ = require('underscore');
+
+module.exports = function(dialect) {
+	var parentValueBlock = dialect.blocks.get('value');
+
+	dialect.blocks.set('value', function(params) {
+		var value = params.value;
+
+		var result;
+		if (_.isBoolean(value)) {
+			result = String(Number(value));
+		} else {
+			result = parentValueBlock(params);
+		}
+
+		return result;
+	});
+
+	dialect.blocks.set('limit', function(params) {
+		var result = '';
+		if (_.isUndefined(params.offset)) {
+			result = 'top(' + dialect.builder._pushValue(params.limit) + ')';
+		}
+		return result;
+	});
+
+	dialect.blocks.set('offset', function(params) {
+		var prefix = (!params.sort) ? 'order by 1 ' : '';
+		var offset = 'offset ' + dialect.builder._pushValue(params.offset) + ' ';
+		var limit = '';
+		var suffix = 'rows';
+
+		if (params.limit) {
+			limit = 'rows fetch next ' + dialect.builder._pushValue(params.limit) + ' ';
+			suffix = 'rows only';
+		}
+
+		return prefix + offset + limit + suffix;
+	});
+
+	dialect.blocks.set('returning', function(params) {
+		var result = dialect.buildBlock('fields', {fields: params.returning});
+
+		if (result) result = 'output ' + result;
+
+		return result;
+	});
+
+	dialect.blocks.set('insert:values', function(params) {
+		var values = params.values;
```

Comment:
https://github.com/2do2go/json-sql/pull/26#discussion_r116426366

This function was refactored a little bit, as was the **`base`** `blocks.js`. This cleans up the `values` and `fields` variable assignment.

`returning` is needed here, because it is used in the next `buildTemplate` call. `{returning}` is now part of the `insertValues` template. See [template](https://github.com/2do2go/json-sql/pull/33/files#diff-5afb54430c401d5f77a4fa01c3dfb201R72)

https://github.com/2do2go/json-sql/pull/26#discussion_r116426456
The unnecessary code was removed